File: ios/Sources/AstrovoxAI/ViewModels/ConversationsViewModel.swift
import Foundation
import Combine

@MainActor
class ConversationsViewModel: ObservableObject {
    @Published var conversations: [Conversation] = []
    @Published var isLoading: Bool = false
    @Published var error: String?

    private let api = APIService()

    func loadConversations() async {
        isLoading = true
        error = nil
        do {
            let remote = try await api.getConversations()
            conversations = remote.map { remoteConversation in
                Conversation(
                    id: remoteConversation.id,
                    title: remoteConversation.title,
                    createdAt: remoteConversation.createdAt,
                    updatedAt: remoteConversation.updatedAt,
                    model: remoteConversation.model
                )
            }
        } catch {
            self.error = error.localizedDescription
        }
        isLoading = false
    }
}

File: android/app/src/main/java/ai/astrovox/presentation/conversations/ConversationsViewModel.kt
package ai.astrovox.presentation.conversations

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import ai.astrovox.data.repository.ChatRepository
import ai.astrovox.domain.model.Conversation
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch

data class ConversationsState(
    val conversations: List<Conversation> = emptyList(),
    val isLoading: Boolean = false,
    val error: String? = null
)

class ConversationsViewModel(private val repository: ChatRepository) : ViewModel() {
    private val _state = MutableStateFlow(ConversationsState())
    val state: StateFlow<ConversationsState> = _state

    init {
        loadConversations()
    }

    fun loadConversations() {
        viewModelScope.launch {
            _state.value = _state.value.copy(isLoading = true)
            try {
                repository.getConversations().collect { list ->
                    _state.value = _state.value.copy(
                        conversations = list.map { entity ->
                            Conversation(
                                id = entity.id,
                                title = entity.title,
                                createdAt = entity.createdAt,
                                updatedAt = entity.updatedAt,
                                model = entity.model
                            )
                        },
                        isLoading = false
                    )
                }
            } catch (e: Exception) {
                _state.value = _state.value.copy(
                    error = e.message ?: "Unknown error",
                    isLoading = false
                )
            }
        }
    }
}
